toplevel/toploop.ml

```diff
@@ -384,6 +379,11 @@ let process_phrases ppf snap phrs =
         (fun () -> List.iter process rest)
     end
 
+let initialize_toplevel_env_or_exit ppf =
+  try initialize_toplevel_env ()
+  with Env.Error _ | Typetexp.Error _ as exn ->
+    Location.report_exception ppf exn; raise (Compenv.Exit_with_status 2)
```

Comment:
Note that I find the location of this function slightly awkward now because its only user is the `prepare` function quite further below.

Reply:
Good point, I inlined the function.